compiler/rustc_hir_analysis/src/hir_ty_lowering/object_safety.rs

```diff
@@ -141,9 +141,8 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                         // `trait_object_dummy_self`, so check for that.
                         let references_self = match pred.skip_binder().term.unpack() {
                             ty::TermKind::Ty(ty) => ty.walk().any(|arg| arg == dummy_self.into()),
-                            ty::TermKind::Const(c) => {
-                                c.ty().walk().any(|arg| arg == dummy_self.into())
-                            }
+                            // FIXME(associated_const_equality): We should walk the const instead of not doing anything
+                            ty::TermKind::Const(_) => false,
```

Comment:
Yes, indeed. You can leave it as is. I need to work on assoc const eq inside trait obj tys anyway. Adding one more FIXME is fine.

I've been meaning to fix `trait_object_dummy_self` getting leaked via const projections (* #123456, #123140, #125957) for a while now (but it's ugly).